src/main/java/sg/ncl/MainController.java

```diff
@@ -3775,4 +3884,41 @@ private String getUsageStatisticsByTeamId(String id) {
         }
         return response.getBody().toString();
     }
+
+    private TeamQuota extractTeamQuotaInfo(String responseBody) {
+        JSONObject object = new JSONObject(responseBody);
+        TeamQuota teamQuota = new TeamQuota();
+        Double charges = Double.parseDouble(accountingProperties.getCharges());
+
+        // amountUsed from SIO will never be null => not checking for null value
+        String usage = object.getString("usage");                 // getting usage in String
+        BigDecimal amountUsed = new BigDecimal(usage);                //  using BigDecimal to handle currency
+        amountUsed = amountUsed.multiply(new BigDecimal(charges));   // usage X charges
+
+        //quota passed from SIO can be null , so we have to check for null value
+        if (object.has("quota")) {
+            Object budgetObject = object.optString("quota", null);
+            if (budgetObject == null) {
+                teamQuota.setBudget("");                  // there is placeholder here
+                teamQuota.setResourcesLeft("Unlimited"); // not placeholder so can pass string over
+            } else {
+                Double budgetInDouble = object.getDouble("quota");          // retrieve budget from SIO in Double
+                BigDecimal budgetInBD = BigDecimal.valueOf(budgetInDouble);     // handling currency using BigDecimal
+
+                // calculate resoucesLeft
+                BigDecimal resourceLeftInBD = budgetInBD.subtract(amountUsed);
+                resourceLeftInBD = resourceLeftInBD.divide(new BigDecimal(charges), 2, BigDecimal.ROUND_HALF_UP);
+
+                // set budget and resourceLeft
+                budgetInBD = budgetInBD.setScale(2, BigDecimal.ROUND_HALF_UP);
+                teamQuota.setBudget(budgetInBD.toString());
+                teamQuota.setResourcesLeft(resourceLeftInBD.toString());
```

Comment:
Round down to its nearest integer and make sure that this value is not smaller than 0.